pkg/binary/envoy/debug/pii_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2019 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debug
+
+import (
+	"testing"
+
+	"bitbucket.org/creachadair/shell"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestProcess(t *testing.T) {
+	format := "[%START_TIME%] \"%REQ(:METHOD)% %REQ(X-ENVOY-ORIGINAL-PATH?:PATH)%" +
+		" %PROTOCOL%\" %RESPONSE_CODE% %RESPONSE_FLAGS% \"%DYNAMIC_METADATA(istio.mixer:status)%\"" +
+		" \"%UPSTREAM_TRANSPORT_FAILURE_REASON%\" %BYTES_RECEIVED% %BYTES_SENT% %DURATION% %RESP(X-ENVOY-UPSTREAM-SERVICE-TIME)%" +
+		" \"%REQ(X-FORWARDED-FOR)%\" \"%REQ(USER-AGENT)%\" \"%REQ(X-REQUEST-ID)%\" \"%REQ(:AUTHORITY)%\" \"%UPSTREAM_HOST%\"" +
+		" %UPSTREAM_CLUSTER% %UPSTREAM_LOCAL_ADDRESS% %DOWNSTREAM_LOCAL_ADDRESS% %DOWNSTREAM_REMOTE_ADDRESS% %REQUESTED_SERVER_NAME%\n"
+
+	formats, ok := shell.Split(format)
+	if !ok {
+		t.Fatalf("error in splitting the format: %s", format)
+	}
+
+	validLog := "[2019-09-05T17:53:36.908Z] \"GET /productpage HTTP/1.1\" 200" +
```

Comment:
You can use string literals with ` to avoid having to double-escape the strings here:
```go
validLog := `[2019-09-05T17:53:36.908Z] "GET /productpage HTTP/1.1" 200` +
...
```

Reply:
And note that max string length in golang is 110 characters (enforced by our linters), so these strings can be a decent bit longer before you wrap.